src/helpers/rerun-pr-checks.ts

```diff
@@ -0,0 +1,55 @@
+/*
+Copyright 2021 Expedia, Inc.
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    https://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import * as core from '@actions/core';
+import { context } from '@actions/github';
+import { octokit } from '../octokit';
+
+export const rerunPrChecks = async () => {
+  /** grab owner in case of fork branch */
+  const {
+    data: {
+      head: {
+        user: { login: owner }
+      }
+    }
+  } = await octokit.pulls.get({
+    pull_number: context.issue.number,
+    ...context.repo
+  });
+  const branch = context.ref.replace('refs/heads/', '');
+  const workflowRuns = await octokit.actions.listWorkflowRunsForRepo({
+    branch,
+    ...context.repo,
+    owner,
+    event: 'pull_request',
+    per_page: 100
+  });
+  if (!workflowRuns.data.total_count) {
+    core.info(`No workflow runs found on branch ${branch} on ${owner}/${context.repo.repo}`);
+    return;
+  }
+  /** grab only latest occurrence of each workflow run */
+  const latestHash = workflowRuns.data.workflow_runs[0].head_sha;
+  const latestRuns = workflowRuns.data.workflow_runs.filter(({ head_sha }) => head_sha === latestHash);
```

Comment:
A bit confused by this part - if you update the test to assert which ids are passed in I'll take another look. Also I slightly prefer `.find(Boolean)` to `[0]`, but if you wanna go with `[0]` let's make it `[0]?.head_sha`

Reply:
`[0]` _should_ always be accessible given that the previous condition returns if `total_count === 0`. but i'll add `?.` for safety just in case GitHub goofs in its response somehow

Reply:
actually, to make this clearer, i think a more elegant way to grab the latest commit hash would just be to do so from the metadata returned from `pulls.get()`. i'll update it to reflect that